src/Sylius/Bundle/CoreBundle/Validator/Constraints/ChannelCodeCollectionValidator.php

```diff
@@ -38,10 +43,46 @@ public function validate(mixed $value, Constraint $constraint): void
             throw new UnexpectedValueException($value, 'array');
         }
 
+        if ($constraint->validateAgainstAllChannels) {
+            $this->validateInChannelCollection($value, $this->channelRepository->findAll(), $constraint);
+
+            return;
+        }
+
+        $object = $this->context->getObject();
+
+        if (property_exists($object, 'promotion')) {
+            $object = $this->propertyAccessor->getValue($object, 'promotion');
```

Comment:
This should be more configurable, without using `if`.  Please add a new property like `parentPropertyPath` to `ChannelCodeCollection` and use this parameter here. This parameter can then be configured in the XML configuration.